addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -87,15 +102,28 @@ def _check_validity_dates(self):
                htype.validity_start > htype.validity_stop:
                 raise ValidationError(_("End of validity period should be greater than start of validity period"))
 
+    @api.multi
+    @api.constrains('balance_limit')
```

Comment:
`accrual` should be in the depends too. So be sure to reset `balance_limit` when unchecking accrual.